yalla-coop/chiltern-music-therapy
# 49 email setup

#### Issues this PR relates to
#49 


=======

_NOTE: REMEMBER NOT TO REVIEW UNLESS ALL ACCEPTANCE CRITERIA HAS BEEN TICKED OFF_

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install
add these values to .env, you can get the them from #1

```
APP_URL
SENDER_EMAIL
SENDER_PASSWORD
```

**to test**:
- go to file `/home/ramy/chiltern-music-therapy/server/modules/user/use-cases/client-signup.js`
- add this code at line 13 and save file, you should see the emails details on the console
- if you want to receive real emails, go to `/home/ramy/chiltern-music-therapy/server/services/emails/index.js` and comment line 29, and set your email for client1 / therapist1 and build db
```js
// to test sign up invitation
events.emit(events.types.THERAPIST_CLIENT.CLIENT_INVITED, {
  therapistClientId: 1,
});

// to test create programme 
events.emit(events.types.PROGRAMME.CREATED, {
  programmeId: 1,
});

// to test update programme 
events.emit(events.types.PROGRAMME.UPDATED, {
  programmeId: 1,
});

// to test create feedback 
events.emit(events.types.PROGRAMME.FEEDBACK.CREATED, {
  feedbackId: 1,
});

// to test create progress 
events.emit(events.types.PROGRESS_UPDATE.CREATED, {
  progressId: 1,
});

// to test response to progress 
events.emit(events.types.PROGRESS_UPDATE.THERAPIST_RESPOND, {
  progressId: 1,
});

```
---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist

Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)

- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs
